### PR TITLE
Enabled AppInsights integration

### PR DESCRIPTION
**Pull request by pravagar**
- Filter events are changed as there are new filter categories.
- Added open in vs code event.

### CODE REVIEW

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
+  }
+
+  outboundDeployClick(template) {
+    let eventData = this.getDeployClickEventDataToLog(template);
+    trackEvent("/sample/deploy/agree", eventData);
+  }
+
+  getDeployClickEventDataToLog(template) {
+    return {
+      template: template
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: please also include the id in the event data

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
       encodeURIComponent(template)
     );
   }
 
+  outboundRepoClick(repository, template) {
+    let eventData = this.getDeployClickEventDataToLog(repository, template);
+    trackEvent("/sample/source", eventData);
+  }
+
+  getDeployClickEventDataToLog(repository, template) {
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: need not define separate methods for event data. all the events can have the same data

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js [hunk 3]
```diff
@@ -1,19 +1,61 @@
...
+
   getDeployLink(template) {
     return (
       "https://portal.azure.com/#create/Microsoft.Template/uri/" +
       encodeURIComponent(template)
     );
   }
 
+  outboundRepoClick(repository, template) {
+    let eventData = this.getDeployClickEventDataToLog(repository, template);
```
Comment (msnehagup) [inline ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: can you use props directly instead of passing params?

Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: It is not a good idea to make a method access member data  if can be avoided.

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js (hunk 2 above)
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: May be in future, different events will need different data

File: ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js (hunk 1 above)
Comment (pravagar) [reply ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js]: in all events? If emit Id, no need to emit enything else. What is ur view?

**Pull request merged**